server/mqtt.go

```diff
@@ -2678,11 +2678,8 @@ func (sess *mqttSession) clear() error {
 		err := sess.jsa.deleteMsg(mqttSessStreamName, seq, true)
 		// Ignore the various errors indicating that the message (or sequence)
 		// is already deleted, can happen in a cluster.
-		if isErrorOtherThan(err, JSSequenceNotFoundErrF) {
-			if !IsNatsErr(err, JSStreamMsgDeleteFailedF) ||
-				!strings.HasSuffix(err.Error(), "no message found (10057)") {
-				return fmt.Errorf("unable to delete session %q record at sequence %v: %v", id, seq, err)
-			}
+		if isErrorOtherThan(err, JSSequenceNotFoundErrF) && isErrorOtherThan(err, JSNoMessageFoundErr) {
```

Comment:
Ok, I see what you are saying regarding low level store API not returning the JS error we want. So maybe revert, but I would then change the `!strings.HasSuffix(...` with `!strings.Contains(err.Error(), ErrStoreMsgNotFound.Error())`, so at least if that error message text were to change, this code would still catch this.